From 4c674608aaaac73d9333f869941a56b28fc47d22 Mon Sep 17 00:00:00 2001
From: tsujan <tsujan2000@gmail.com>
Date: Sat, 10 Apr 2021 17:39:47 +0430
Subject: [PATCH] Check pressed mouse buttons before closing grouped task popup
 (#1601)

The lack of checking didn't have an effect with the official Qt releases but it causes a bug with the latest official patches. Logically, the pressed mouse buttons should have always been checked.

Closes https://github.com/lxqt/lxqt-panel/issues/1600
---
 CHANGELOG                        | 1 +
 plugin-taskbar/lxqttaskgroup.cpp | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/plugin-taskbar/lxqttaskgroup.cpp b/plugin-taskbar/lxqttaskgroup.cpp
index 7f2367ce1..f1db0f179 100644
--- a/plugin-taskbar/lxqttaskgroup.cpp
+++ b/plugin-taskbar/lxqttaskgroup.cpp
@@ -576,7 +576,8 @@ void LXQtTaskGroup::dragLeaveEvent(QDragLeaveEvent *event)
 void LXQtTaskGroup::mouseMoveEvent(QMouseEvent* event)
 {
     // if dragging the taskgroup, do not show the popup
-    setPopupVisible(false, true);
+    if (event->buttons() & Qt::LeftButton)
+        setPopupVisible(false, true);
     LXQtTaskButton::mouseMoveEvent(event);
 }
 
